### PR TITLE
refactor: fine-grained github actions steps

### PR DESCRIPTION
**Pull request by WuShaoling**
Changelog: 
- GitHub Actions: split the Test step into multiple steps
- Split the submodule `md-lint` as an independent service:
   GitHub: [open-md-checker](https://github.com/WuShaoling/open-md-checker)
   npmjs: [open-md-checker(https://www.npmjs.com/package/open-md-checker)

- Split the submodule `license-check` as an independent service:
   Github: [open-license-check](https://github.com/WuShaoling/open-license-check)
   Docker Image: [openxlab/open-license-check](https://hub.docker.com/repository/docker/openxlab/open-license-check)

- `md-lint` command reserved for local testing
- `license-check` command reserved for local testing

### CODE REVIEW

File: .github/workflows/hypertrons.yml [hunk 1]
```diff
@@ -12,39 +12,47 @@ jobs:
...
-      env:
-        SLACK_COLOR: '#FF0000'
-        SLACK_MESSAGE: ${{ env.SLACK_MESSAGE_TEMPLATE}} `failure`.
-    - name: Slack Notification(Cancelled)
-      uses: rtCamp/action-slack-notify@v2.0.0
-      if: cancelled() && github.event_name == 'push' && github.repository == 'hypertrons/hypertrons'
-      env:
-        SLACK_COLOR: '#FFFF00'
-        SLACK_MESSAGE: ${{ env.SLACK_MESSAGE_TEMPLATE}} `cancelled`.
+      - name: Checkout
```
Comment (frank-zsy) [inline .github/workflows/hypertrons.yml]: The indent has been changed, does both format is right for yml file?

Comment (WuShaoling) [reply .github/workflows/hypertrons.yml]: The indentation here does not affect the syntax. The demos provided by GitHub are not indented. The demos provided by the YAML official website is indented. The indentation used here conforms to the official syntax, and it looks more hierarchical.

**Comment by frank-zsy** [comment]
@WuShaoling PR conflict.

**Comment by frank-zsy** [comment]
Gonna merge by hand due to action files can not be merged by apps.

**Pull request merged**